includes/classes/InternalConnections/NetworkSiteConnection.php

```diff
@@ -934,4 +940,53 @@ public static function wpseo_og_url( $og_url ) {
 
 		return $og_url;
 	}
+
+	/**
+	 * Updates a post content via a REST request after the new post is created
+	 * in order to get the rendered content.
+	 *
+	 * @param int $new_post_id The new post ID that was pulled.
+	 * @return void
+	 */
+	public function update_content_via_rest( $new_post_id ) {
+
+		$post = get_post( $new_post_id );
+		if ( ! is_a( $post, '\WP_Post' ) ) {
+			return;
+		}
+
+		$original_blog_id = absint( get_post_meta( $post->ID, 'dt_original_blog_id', true ) );
+		$original_post_id = absint( get_post_meta( $post->ID, 'dt_original_post_id', true ) );
+
+		$rest_url = false;
+		if ( ! empty( $original_blog_id ) && ! empty( $original_post_id ) ) {
+			$rest_url = Utils\get_rest_url( $original_blog_id, $original_post_id );
+		}
+
+		if ( empty( $rest_url ) ) {
+			return;
+		}
+
+		$request = apply_filters( 'dt_update_content_via_request_args', [], $new_post_id, $this );
+
+		$response = wp_remote_get( $rest_url, $request );
+
+		$body = wp_remote_retrieve_body( $response );
```

Comment:
We'll want to add a few more checks here, just to be safe. Basically check the `$response` value isn't a WP Error and check that the response code (`wp_remote_retrieve_response_code`) matches what we want. Can see examples of that here: https://github.com/10up/distributor/blob/develop/includes/classes/ExternalConnections/WordPressExternalConnection.php#L200

Reply:
FYI, `wp_remote_retrieve_body()` already checks for WP_Error.

Reply:
Ah, didn't realize that. I still like the explicit check, so we can output a different error message in each instance, to better help with debugging issues.